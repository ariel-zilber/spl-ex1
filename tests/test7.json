{
  "parties": [
    {
      "name": "The Lannisters",
      "mandates": 70,
      "join_policy": "L"
    },
    {
      "name": "The Baratheons",
      "mandates": 30,
      "join_policy": "L"
    },
    {
      "name": "The Targaryens",
      "mandates": 10,
      "join_policy": "L"
    },
    {
      "name": "The Starks",
      "mandates": 10,
      "join_policy": "L"
    }
  ],
  "graph": [
    [0, 1, 2, 3],
    [1, 0, 4, 5],
    [2, 4, 0, 6],
    [3, 5, 6, 0]
  ],
  "agents": [
    {
      "party_id": 2,
      "selection_policy": "E"
    },
    {
      "party_id": 3,
      "selection_policy": "E"
    }

  ]
}
  