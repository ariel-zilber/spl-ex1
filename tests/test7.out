[
    {
        "agents": [
            {
                "id": 0,
                "partyId": 2
            },
            {
                "id": 1,
                "partyId": 3
            }
        ],
        "coalitions": [
            {
                "mandates": 10,
                "parties": [
                    2
                ]
            },
            {
                "mandates": 10,
                "parties": [
                    3
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 2
            },
            {
                "from": 0,
                "to": 3,
                "weight": 3
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 4
            },
            {
                "from": 1,
                "to": 3,
                "weight": 5
            },
            {
                "from": 2,
                "to": 0,
                "weight": 2
            },
            {
                "from": 2,
                "to": 1,
                "weight": 4
            },
            {
                "from": 2,
                "to": 3,
                "weight": 6
            },
            {
                "from": 3,
                "to": 0,
                "weight": 3
            },
            {
                "from": 3,
                "to": 1,
                "weight": 5
            },
            {
                "from": 3,
                "to": 2,
                "weight": 6
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 70,
                "name": "The Lannisters",
                "state": "Waiting"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Baratheons",
                "state": "Waiting"
            },
            {
                "id": 2,
                "mandates": 10,
                "name": "The Targaryens",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 10,
                "name": "The Starks",
                "state": "Joined"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 2
            },
            {
                "id": 1,
                "partyId": 3
            }
        ],
        "coalitions": [
            {
                "mandates": 10,
                "parties": [
                    2
                ]
            },
            {
                "mandates": 10,
                "parties": [
                    3
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 2
            },
            {
                "from": 0,
                "to": 3,
                "weight": 3
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 4
            },
            {
                "from": 1,
                "to": 3,
                "weight": 5
            },
            {
                "from": 2,
                "to": 0,
                "weight": 2
            },
            {
                "from": 2,
                "to": 1,
                "weight": 4
            },
            {
                "from": 2,
                "to": 3,
                "weight": 6
            },
            {
                "from": 3,
                "to": 0,
                "weight": 3
            },
            {
                "from": 3,
                "to": 1,
                "weight": 5
            },
            {
                "from": 3,
                "to": 2,
                "weight": 6
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 70,
                "name": "The Lannisters",
                "state": "Waiting"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Baratheons",
                "state": "CollectingOffers"
            },
            {
                "id": 2,
                "mandates": 10,
                "name": "The Targaryens",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 10,
                "name": "The Starks",
                "state": "Joined"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 2
            },
            {
                "id": 1,
                "partyId": 3
            }
        ],
        "coalitions": [
            {
                "mandates": 10,
                "parties": [
                    2
                ]
            },
            {
                "mandates": 10,
                "parties": [
                    3
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 2
            },
            {
                "from": 0,
                "to": 3,
                "weight": 3
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 4
            },
            {
                "from": 1,
                "to": 3,
                "weight": 5
            },
            {
                "from": 2,
                "to": 0,
                "weight": 2
            },
            {
                "from": 2,
                "to": 1,
                "weight": 4
            },
            {
                "from": 2,
                "to": 3,
                "weight": 6
            },
            {
                "from": 3,
                "to": 0,
                "weight": 3
            },
            {
                "from": 3,
                "to": 1,
                "weight": 5
            },
            {
                "from": 3,
                "to": 2,
                "weight": 6
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 70,
                "name": "The Lannisters",
                "state": "CollectingOffers"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Baratheons",
                "state": "CollectingOffers"
            },
            {
                "id": 2,
                "mandates": 10,
                "name": "The Targaryens",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 10,
                "name": "The Starks",
                "state": "Joined"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 2
            },
            {
                "id": 1,
                "partyId": 3
            }
        ],
        "coalitions": [
            {
                "mandates": 10,
                "parties": [
                    2
                ]
            },
            {
                "mandates": 10,
                "parties": [
                    3
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 2
            },
            {
                "from": 0,
                "to": 3,
                "weight": 3
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 4
            },
            {
                "from": 1,
                "to": 3,
                "weight": 5
            },
            {
                "from": 2,
                "to": 0,
                "weight": 2
            },
            {
                "from": 2,
                "to": 1,
                "weight": 4
            },
            {
                "from": 2,
                "to": 3,
                "weight": 6
            },
            {
                "from": 3,
                "to": 0,
                "weight": 3
            },
            {
                "from": 3,
                "to": 1,
                "weight": 5
            },
            {
                "from": 3,
                "to": 2,
                "weight": 6
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 70,
                "name": "The Lannisters",
                "state": "CollectingOffers"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Baratheons",
                "state": "CollectingOffers"
            },
            {
                "id": 2,
                "mandates": 10,
                "name": "The Targaryens",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 10,
                "name": "The Starks",
                "state": "Joined"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 2
            },
            {
                "id": 1,
                "partyId": 3
            },
            {
                "id": 2,
                "partyId": 1
            }
        ],
        "coalitions": [
            {
                "mandates": 10,
                "parties": [
                    2
                ]
            },
            {
                "mandates": 40,
                "parties": [
                    3,
                    1
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 2
            },
            {
                "from": 0,
                "to": 3,
                "weight": 3
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 4
            },
            {
                "from": 1,
                "to": 3,
                "weight": 5
            },
            {
                "from": 2,
                "to": 0,
                "weight": 2
            },
            {
                "from": 2,
                "to": 1,
                "weight": 4
            },
            {
                "from": 2,
                "to": 3,
                "weight": 6
            },
            {
                "from": 3,
                "to": 0,
                "weight": 3
            },
            {
                "from": 3,
                "to": 1,
                "weight": 5
            },
            {
                "from": 3,
                "to": 2,
                "weight": 6
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 70,
                "name": "The Lannisters",
                "state": "CollectingOffers"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Baratheons",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 10,
                "name": "The Targaryens",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 10,
                "name": "The Starks",
                "state": "Joined"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 2
            },
            {
                "id": 1,
                "partyId": 3
            },
            {
                "id": 2,
                "partyId": 1
            },
            {
                "id": 3,
                "partyId": 0
            }
        ],
        "coalitions": [
            {
                "mandates": 10,
                "parties": [
                    2
                ]
            },
            {
                "mandates": 110,
                "parties": [
                    3,
                    1,
                    0
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 2
            },
            {
                "from": 0,
                "to": 3,
                "weight": 3
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 4
            },
            {
                "from": 1,
                "to": 3,
                "weight": 5
            },
            {
                "from": 2,
                "to": 0,
                "weight": 2
            },
            {
                "from": 2,
                "to": 1,
                "weight": 4
            },
            {
                "from": 2,
                "to": 3,
                "weight": 6
            },
            {
                "from": 3,
                "to": 0,
                "weight": 3
            },
            {
                "from": 3,
                "to": 1,
                "weight": 5
            },
            {
                "from": 3,
                "to": 2,
                "weight": 6
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 70,
                "name": "The Lannisters",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Baratheons",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 10,
                "name": "The Targaryens",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 10,
                "name": "The Starks",
                "state": "Joined"
            }
        ]
    }
]
